findatastructures/src/finmap/mod.rs

```diff
@@ -0,0 +1,28 @@
+pub trait FinMap: PartialEq + Clone {
```

Comment:
What is this trait doing? Please add comments.

Reply:
Also, the trait bounds here are not necessary and only restrict how it can be used.